src/core/i18n/translator.js

```diff
@@ -11,13 +13,42 @@ export default class Translator {
     return this._interpolate(stringToInterpolate, interpolationParams);
   }
 
+  /**
+   * If translations is json, parse it and choose the correct plural form.
+   * Otherwise just return the non-interpolated translation string.
+   * @param {string} translations
+   * @param {number} count
+   * @returns {string}
+   */
   static _selectPluralization (translations, count) {
     try {
       translations = JSON.parse(translations);
-      return count > 1 ? translations.plural : translations['1'];
     } catch (e) {
       return translations;
     }
+    return this._selectPluralForm(translations, count);
+  }
+
+  /**
+   * Returns the correct plural form given a parsed translations object and count.
+   * @param {Object} parsedTranslations
+   * @param {number} count
+   * @returns {string}
+   */
+  static _selectPluralForm (parsedTranslations, count) {
+    const locale = parsedTranslations.locale;
+    const oneToNArray = this._generateArrayOneToN(locale);
+    const pluralFormIndex = getPluralFunc(locale)(count, oneToNArray);
+    return parsedTranslations[pluralFormIndex];
```

Comment:
Thoughts on utilizing the hasLang function from plural-forms to validate the locale?

Reply:
sure I kind of like that. I don't hope we don't ever need this but we did have a case in the past where some client wanted to use a "fake" locale in our pages system